optimum/intel/openvino/modeling_diffusion.py

```diff
@@ -838,17 +846,23 @@ def reshape(
 
         if self.text_encoder is not None:
             self.text_encoder.model = self._reshape_text_encoder(
-                self.text_encoder.model, batch_size, self.tokenizer.model_max_length
+                self.text_encoder.model,
+                batch_size,
+                self.tokenizer.model_max_length if "Gemma" not in self.tokenizer.__class__.__name__ else -1,
             )
 
         if self.text_encoder_2 is not None:
             self.text_encoder_2.model = self._reshape_text_encoder(
-                self.text_encoder_2.model, batch_size, self.tokenizer_2.model_max_length
+                self.text_encoder_2.model,
+                batch_size,
+                self.tokenizer_2.model_max_length if "Gemma" not in self.tokenizer.__class__.__name__ else -1,
             )
 
         if self.text_encoder_3 is not None:
             self.text_encoder_3.model = self._reshape_text_encoder(
-                self.text_encoder_3.model, batch_size, self.tokenizer_3.model_max_length
+                self.text_encoder_3.model,
+                batch_size,
+                self.tokenizer_3.model_max_length if "Gemma" not in self.tokenizer.__class__.__name__ else -1,
```

Comment:
the check should be `if "Gemma" not in self.tokenizer_3.__class__.__name__` ?
```suggestion
                getattr(self.tokenizer_3, "model_max_length", -1),
```

Reply:
sana will file on reshaping stage as model_max_length too large and can not be used for reshaping

Reply:
what about `if "Gemma" not in self.tokenizer.__class__.__name__` should be the same in all three checks ?

Reply:
not enough data to verify yet as we have only 1 model where gemma used and it has only 1 text_encoder, so I'll return back fro text_encoder_2 and 3

Reply:
updated